javalib/src/main/scala/java/util/concurrent/ConcurrentLinkedQueue.scala

```diff
@@ -0,0 +1,45 @@
+package java.util.concurrent
+
+import java.util._
+
+class ConcurrentLinkedQueue[E]()
+    extends AbstractQueue[E] with Queue[E] with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  private var inner: LinkedList[E] = new LinkedList()
+
+  override def add(e: E): Boolean =
+    if (e == null) throw new NullPointerException()
+    else inner.add(e)
+
+  override def offer(e: E): Boolean =
+    add(e)
+
+  override def poll(): E =
+    inner.poll()
+
+  override def peek(): E =
+    inner.peek()
+
+  override def isEmpty(): Boolean =
+    inner.isEmpty()
+
+  override def size(): Int =
+    inner.size()
```

Comment:
I will double check but the method signature returns an ```int ``` in java, so probably also the size in  java will overflow, I will check that this evening